Cython/Compiler/ModuleNode.py

```diff
@@ -2299,6 +2299,13 @@ def generate_module_init_func(self, imported_modules, env, code):
 
         code.put_declare_refcount_context()
         code.putln("#if CYTHON_PEP489_MULTI_PHASE_INIT")
+        code.putln('PyObject *moddict, *spec;')
+        code.putln('moddict = PyModule_GetDict(__pyx_pyinit_module);')
+        code.putln('spec = PyDict_GetItemString(moddict, "__spec__");')
```

Comment:
These calls lack error handling. And I'd also prefer having this setup in an external C helper function (as before), rather than generating it line by line. It's just easier to maintain that way.